src/Symfony/Component/Validator/Constraints/ExpressionValidator.php

```diff
@@ -46,6 +47,10 @@ public function validate($value, Constraint $constraint)
         $variables['value'] = $value;
         $variables['this'] = $this->context->getObject();
 
+        if (strlen($constraint->dataPath) > 0) {
+            $variables['this'] = $this->getPropertyAccessor()->getValue($this->context, $constraint->dataPath);
```

Comment:
In the other hand, if Symfony deciders decides to make it configurable I think we could start the path from the `root` i.e.:
```php
$this->getPropertyAccessor()->getValue($this->context->getRoot(), $constraint->dataPath);
```
else, I suggest pass the `root` value to `this` directly if it's `null` i.e::
```php
if (null === $variables['this'] = $this->context->getObject()) {
    $variables['this'] = $this->context->getRoot();
}
```
thus, you don't have to worry about the "dataPath", you know the data linked to the root, therefore you just have to access them:
```php
new Expression(array('expression' => 'value <= this["data"]["dateEnd"]'));
```
wdyt?